crates/bitwarden-auth/src/identity/login_via_password/mod.rs

```diff
@@ -0,0 +1,94 @@
+//! Password-based authentication for Bitwarden users.
+//!
+//! This module implements the password login flow, which requires two steps:
+//!
+//! 1. **Prelogin**: Retrieve the user's KDF configuration with
+//!    [`LoginClient::get_password_prelogin`]
+//! 2. **Login**: Authenticate with [`LoginClient::login_via_password`] using the KDF settings
+//!
+//! # Security Model
+//!
+//! The master password is **never sent to the server**. Instead:
+//! - User's KDF settings (PBKDF2 or Argon2id) are fetched during prelogin
+//! - Master password is hashed locally using these settings
+//! - Only the derived hash is transmitted for authentication
```

Comment:
This suggests that the output of the KDF is sent to the server. That is not what is happening. 

The KDF output - raw masterkey - is:
1. Stretched into an AES256_CBC_HMAC encryption key to unwrap the userkey
2. Hashed again with a single round of PDKBF2 with the password as salt to get the "serverAuthenticationHash" <- this is sent to the server

Reply:
Sorry about that. Fixed with: dd0db639b281644f02e8d89eff15609fd13b678c